app/src/main/java/org/schabi/newpipe/settings/NotificationsSettingsFragment.kt

```diff
@@ -0,0 +1,114 @@
+package org.schabi.newpipe.settings
+
+import android.content.SharedPreferences
+import android.content.SharedPreferences.OnSharedPreferenceChangeListener
+import android.os.Bundle
+import androidx.core.content.ContextCompat
+import androidx.preference.Preference
+import com.google.android.material.snackbar.Snackbar
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.disposables.Disposable
+import org.schabi.newpipe.R
+import org.schabi.newpipe.database.subscription.NotificationMode
+import org.schabi.newpipe.database.subscription.SubscriptionEntity
+import org.schabi.newpipe.error.ErrorActivity
+import org.schabi.newpipe.error.ErrorInfo
+import org.schabi.newpipe.error.UserAction
+import org.schabi.newpipe.local.subscription.SubscriptionManager
+import org.schabi.newpipe.notifications.NotificationHelper
+import org.schabi.newpipe.notifications.NotificationWorker
+import org.schabi.newpipe.notifications.ScheduleOptions
+import java.util.Locale
+
+class NotificationsSettingsFragment : BasePreferenceFragment(), OnSharedPreferenceChangeListener {
+
+    private var notificationWarningSnackbar: Snackbar? = null
+    private var loader: Disposable? = null
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        addPreferencesFromResource(R.xml.notifications_settings)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        defaultPreferences.registerOnSharedPreferenceChangeListener(this)
+    }
+
+    override fun onStop() {
+        defaultPreferences.unregisterOnSharedPreferenceChangeListener(this)
+        super.onStop()
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
+        val context = context ?: return
+        if (key == getString(R.string.streams_notifications_interval_key) || key == getString(R.string.streams_notifications_network_key)) {
+            NotificationWorker.schedule(context, ScheduleOptions.from(context), true)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        val enabled = NotificationHelper.isNotificationsEnabledNative(context)
+        preferenceScreen.isEnabled = enabled
+        if (!enabled) {
+            if (notificationWarningSnackbar == null) {
+                notificationWarningSnackbar = Snackbar.make(
+                    listView,
+                    R.string.notifications_disabled,
+                    Snackbar.LENGTH_INDEFINITE
+                ).apply {
+                    setAction(R.string.settings) { v ->
+                        NotificationHelper.openNativeSettingsScreen(v.context)
+                    }
+                    setActionTextColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.snackbar_action_color
+                        )
+                    )
```

Comment:
Why a different color?

Reply:
Changed to `Color.YELLOW` as elsewhere

Reply:
Just do not use `setActionTextColor` at all and let the color be taken from the theme, as it is done everywhere else throughout the app.